src/goto-programs/remove_ternary.h

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************\
+
+Module: Remove function returns
+
+Author: Daniel Kroening
+
+Date:   September 2009
+
+\*******************************************************************/
+
+#ifndef CPROVER_REMOVE_TERNARY_H
+#define CPROVER_REMOVE_TERNARY_H
+
+#include <goto-programs/goto_model.h>
+
+class remove_ternaryt
+{
+public:
+  explicit remove_ternaryt(symbol_tablet &_symbol_table):
+    symbol_table(_symbol_table)
+  {
+  }
+
+  void operator()(goto_functionst &goto_functions);
+
+protected:
+  symbol_tablet &symbol_table;
+
+  unsigned int replaced = 0;
+
+  void replace_ternary(
+      goto_programt &goto_program,
+      goto_programt::instructionst::iterator &i_it,
+          exprt &expr,
+          bool lhs);
+
+  void contains_ternary(exprt &expr, bool &contains);
+
+  bool contains_ternary(exprt &expr);
```

Comment:
Neither the "contains_ternary" nor "replaced" are used anywhere; the former is really dangerous as it will kill all sharing (well, actually contains_ternary would have to take a const exprt and be const).